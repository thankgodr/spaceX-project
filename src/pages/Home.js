import { Container } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import RocketView from '../components/RocketView';

const Home = () => {
  const state = useSelector((state) => state.rockets);
  return (
        <Container className='mt-2'>
            {state.map((rocket) => <RocketView
                image={rocket.flickr_images[0]}
                id={rocket.id}
                title={rocket.name}
                description={rocket.description}
                key={rocket.id}
                reserved={rocket.reserved}
            />)}
        </Container>
  );
};

export default Home;