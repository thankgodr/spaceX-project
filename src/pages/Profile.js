import {
  Container, Row, Col, ListGroup,
} from 'react-bootstrap';
import { useSelector } from 'react-redux';

const Profile = () => {
  const state = useSelector((state) => state.rockets);
  const missionState = useSelector((state) => state.missions);
  return (
        <Container>
            <Row>
                <Col md={6}>
                <h3>My Missions</h3>
                    <ListGroup>
                        {missionState.filter((missions) => missions.reserved).map((mission) => <ListGroup.Item key={mission.id} className="p-3 text-left">{mission.name}</ListGroup.Item>)}
                    </ListGroup>
                </Col>
                <Col md={6}>
                    <h3>My Rockets</h3>
                    <ListGroup>
                        {state.filter((rockets) => rockets.reserved).map((rocket) => <ListGroup.Item key={rocket.id} className="p-3 text-left">{rocket.name}</ListGroup.Item>)}
                    </ListGroup>
                </Col>

            </Row>
        </Container>

  );
};
export default Profile;