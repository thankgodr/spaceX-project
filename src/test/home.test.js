import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Home from '../pages/Home';
import Profile from '../pages/Profile';
import store from '../redux/configureStore';

describe('test pages snapshot', () => {
  test('Test home snapshot', () => {
    const home = renderer.create(
            <Provider store={store}>
                <Home />
            </Provider>,
    ).toJSON();
    expect(home).toMatchSnapshot();
  });

  test('Test Profile snapshot', () => {
    const profile = renderer.create(
            <Provider store={store}>
                <Profile/>
            </Provider>,
    ).toJSON();
    expect(profile).toMatchSnapshot();
  });
});