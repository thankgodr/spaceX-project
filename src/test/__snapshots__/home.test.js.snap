// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test pages snapshot Test Profile snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <h3>
        My Missions
      </h3>
      <div
        className="list-group"
        onKeyDown={[Function]}
      />
    </div>
    <div
      className="col-md-6"
    >
      <h3>
        My Rockets
      </h3>
      <div
        className="list-group"
        onKeyDown={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`test pages snapshot Test home snapshot 1`] = `
<div
  className="mt-2 container"
/>
`;
